/*
* Brian R Taylor
* brian.taylor@bolderflight.com
* 
* Copyright (c) 2020 Bolder Flight Systems
*/

/*
* This protobuf file describes the data log format, which is transpiled
* to a C header and source file for use in the flight code. There should
* only be a single message defined, DatalogMessage, which defines all
* of the data to be logged. The message structure should be flat
* (i.e. no nested messages), consisting of native types.
*/

syntax = "proto3";

/* Data log message */
message DatalogMessage {
  int32 inceptor_mode0 = 1;
  int32 inceptor_mode1 = 2;
  int32 inceptor_mode2 = 3;
  bool inceptor_throttle_en = 4;
  float inceptor_throttle = 5;
  float inceptor_pitch = 6;
  float inceptor_roll = 7;
  float inceptor_yaw = 8;
  double sys_time_s = 9;
  float imu_accel_x_mps2 = 10;
  float imu_accel_y_mps2 = 11;
  float imu_accel_z_mps2 = 12;
  float imu_gyro_x_radps = 13;
  float imu_gyro_y_radps = 14;
  float imu_gyro_z_radps = 15;
  float imu_mag_x_ut = 16;
  float imu_mag_y_ut = 17;
  float imu_mag_z_ut = 18;
  int32 gnss_fix = 19;
  int32 gnss_num_sats = 20;
  int32 gnss_week = 21;
  float gnss_alt_wgs84_m = 22;
  float gnss_ned_vel_x_mps = 23;
  float gnss_ned_vel_y_mps = 24;
  float gnss_ned_vel_z_mps = 25;
  float gnss_horz_acc_m = 26;
  float gnss_vert_acc_m = 27;
  float gnss_vel_acc_mps = 28;
  double gnss_lat_rad = 29;
  double gnss_lon_rad = 30;
  double gnss_tow_s = 31;
  float airdata_static_pres_pa = 32;
  float airdata_diff_pres_pa = 33;
  float sen_proc_accel_x_mps2 = 34;
  float sen_proc_accel_y_mps2 = 35;
  float sen_proc_accel_z_mps2 = 36;
  float sen_proc_gyro_x_radps = 37;
  float sen_proc_gyro_y_radps = 38;
  float sen_proc_gyro_z_radps = 39;
  float sen_proc_pitch_rad = 40;
  float sen_proc_roll_rad = 41;
  float sen_proc_yaw_rad = 42;
  float sen_proc_static_pres_pa = 43;
  float sen_proc_diff_pres_pa = 44;
  float sen_proc_alt_wgs84_m = 45;
  float sen_proc_alt_pres_m = 46;
  float sen_proc_ias_mps = 47;
  float sen_proc_eas_mps = 48;
  float sen_proc_ned_vel_x_mps = 49;
  float sen_proc_ned_vel_y_mps = 50;
  float sen_proc_ned_vel_z_mps = 51;
  double sen_proc_latitude_rad = 52;
  double sen_proc_longitude_rad = 53;
  repeated float effector_pwm = 54;
  repeated float effector_sbus = 55;
  repeated float aux = 56;
}

